qa/integration/services/service_locator.rb

```diff
@@ -1,19 +1,30 @@
-require_relative "kafka"
-require_relative "elasticsearch"
+# encoding: utf-8
+require_relative "service"
 
 # This is a registry used in Fixtures so a test can get back any service class
 # at runtime
 # All new services should register here
 class ServiceLocator
+  FILE_PATTERN = "_service.rb"
 
   def initialize(settings)
     @services = {}
-    @services["logstash"] = Logstash.new(settings)
-    @services["kafka"] = Kafka.new(settings)
-    @services["elasticsearch"] = ElasticsearchService.new(settings)
+    available_services do |name, klass|
+      @services[name] = klass.new(settings)
+    end
   end
 
   def get_service(name)
-    @services[name]
+    @services.fetch(name)
+  end
+
+  def available_services
```

Comment:
Hash#fetch is awesome we should use it more.. with it its over looking all over the place when you have a nil value.